cpu/qn908x/include/periph_cpu.h

```diff
@@ -149,6 +149,51 @@ enum {
 #define TIMER_MAX_VALUE     (0xffffffff)
 /** @} */
 
+/**
+ * @brief   I2C bus speed values in kbit/s.
+ *
+ * @note    We support arbitrary speed values up to 400 kbit/s.
+ *
+ * @{
+ */
+#define HAVE_I2C_SPEED_T
+typedef enum {
+    I2C_SPEED_LOW       = 10,       /**< low speed mode:          ~10 kbit/s */
+    I2C_SPEED_NORMAL    = 100,      /**< normal mode:            ~100 kbit/s */
+    I2C_SPEED_FAST      = 400,      /**< fast mode:              ~400 kbit/s */
+    I2C_SPEED_FAST_PLUS = 400,      /**< not supported, capped at 400 kbit/s */
+    I2C_SPEED_HIGH      = 400,      /**< not supported, capped at 400 kbit/s */
+} i2c_speed_t;
+/** @} */
+
+/**
+ * @brief   I2C configuration options
+ */
+typedef struct {
+    I2C_Type *dev;          /**< hardware device */
+    gpio_t pin_scl;         /**< SCL pin */
+    gpio_t pin_sda;         /**< SDA pin */
+    uint32_t speed;         /**< bus speed in kbit/s */
```

Comment:
might also use `uint16_t` if 400 is the largest value.
`uint32_t` makes it look like `speed` was in bit/s.

Reply:
I made it uint16_t.